bodhi/tests/server/services/test_releases.py

```diff
@@ -299,3 +346,32 @@ def test_query_releases_html_two_releases_same_state(self):
         self.assertEquals(res.content_type, 'text/html')
         self.assertIn('Fedora 22', res)
         self.assertIn('Fedora 42', res)
+
+    def test_add_packages(self):
+        self.app.put('/releases/F22/critical_packages', {'packages': 'python,json,nodejs,git'},
+                     headers={'Accept': 'application/json'})
+
+        release = self.db.query(Release).filter(Release.name == 'F22').one()
+        pkg_list = [pkg.name for pkg in release.critpath_pkgs]
+
+        self.assertEquals(len(release['critpath_pkgs']), 4)
+        self.assertIn('python', pkg_list)
+        self.assertIn('json', pkg_list)
+        self.assertIn('nodejs', pkg_list)
+        self.assertIn('git', pkg_list)
```

Comment:
We also need a test that ```PUT``` works with releases that already had packages.

Reply:
You could just assert that the set of packages is equal rather than 4 separate assertions.